packages/agent/app/lane.ts

```diff
@@ -12,40 +11,44 @@ const serializeError: (error: ErrorDetails) => ErrorDetails = ({ message, fileNa
   stack
 });
 
-export function *runLane(testRunId: string, agent: Agent, test: TestImplementation, path: string[]): Operation<void> {
+interface TestEvents {
+  send(event: TestEvent): void;
+}
+
+export function *runLane(testRunId: string, agent: TestEvents, test: TestImplementation, path: string[]): Operation<void> {
```

Comment:
Should `agent` be renamed `events`?

```suggestion
export function *runLane(testRunId: string, events: TestEvents, test: TestImplementation, path: string[]): Operation<void> {
```